contrib/util/deidentification/deidentification.php

```diff
@@ -0,0 +1,414 @@
+                  );
+
+/* function to clear the present value if a record's column and replace it with a value if spoecified
+   Input: $con, $table, $column, $value = value to replace with
+*/
+function removeColumn($con, $table, $column, $value = ''){
+    $removeSS = ("Update $table SET $column='$value' where 1 ");
+    $query = mysqli_query($con, $removeSS) or print( "\n QUERY '$removeSS' DID NOT WORK.  PLEASE VERIFY THE TABLE AND COLUMN EXISTS \n");
+    if($query) print("\n Query '$removeSS' completed! \n");
+
+}
+
+/* This function replaces the first and last name of the patient with a auto generated name
+    removes data from the encounter, removes personal information from the patient_data, notes, and form_encounter
+    tables. This should be the first function called.
+
+    INPUT: connection, last name array, firstname arrays, debug
+    OUTPUT: Number of patients DEID.
+*/
+function deIdPatientData($con, $lnames, $male, $female, $DEBUG=false){
+
+    removeColumn($con, "patient_data", "ss", "0000-00-00");
+    removeColumn($con, "form_encounter", "reason", "reason goes here, bucko");
+    removeColumn($con, "form_encounter", "facility", "Service Facility");
+    removeColumn($con, "notes", "note", "notes goes here, Captian");
+
+    $i = 0;
+
+    $removeLname = ("Select lname, pid, id, ss, street, sex from patient_data ");
+    $query = mysqli_query($con, $removeLname);
+    while($result = mysqli_fetch_array($query))
+    {
+        if ($DEBUG===TRUE) {if($i ===10)break;}
+        $i++;
+        $string = '';
+        //Give the user a new last name in patient_data.lname
+        $last_name = $lnames[rand(0,800)];
+
+        //Give the user a new first name
+        $first_name_male = $male[rand(0,32)];
+        $first_name_female = $female[rand(0,74)];
+
+        //Change the street address patient_Data.street
+        $street = rand(1,9999)." ".rand(0,200)." Avenue ";
+
+        //remove the drivers license
+        $drivers_license = rand(2,999).rand(0,999).rand(0,99);
+
+        //change the patient_data.phone_home
+        $phone_home = rand(200,999)."-".rand(200,999)."-".rand(1000,9999);
+
+        $string = "update patient_data set lname = '$last_name', ";
+
+        if($result['sex'] === 'Male'){
+
+            $string .= " fname = '$first_name_male', ";
+        }
+        else{
+            $string .= " fname = '$first_name_female', ";
+        }
+
+        $string .= " drivers_license = '$drivers_license', ";
+
+        $string .= " phone_home = '$phone_home', ";
+
+        $string .= " phone_cell = '$phone_home', ";
+
+        $string .= " phone_biz = '$phone_home', ";
+
+        $string .= " contact_relationship = '', ";
+
+        $string .= " email = '', ";
+
+        $string .= " street = '$street' ";
+
+        $string .= "where pid = "."'".$result['pid']."' ; " ;
```

Comment:
You should probably just preserve PID, or re-order the ID fields, then set PID =ID.